### PR TITLE
fix(hubble): replace tendermint-rpc library

### PR DESCRIPTION
**Pull request by qlp**

### CODE REVIEW

File: hubble/src/indexer/tm/block_handle.rs [hunk 1]
```diff
@@ -21,53 +25,67 @@ use crate::indexer::{
...
-            hash: self.block_id.hash.to_string(),
-            timestamp: self.header.time.into(),
+            height: self.header.height.inner() as u64,
+            hash: self
+                .block_id
+                .hash
+                .ok_or(IndexerError::ProviderError(eyre!("expected hash")))?
+                .to_string(),
+            timestamp: OffsetDateTime::from_unix_timestamp_nanos(
+                self.header.time.as_unix_nanos() as i128
```
Comment (KaiserKarel) [inline hubble/src/indexer/tm/block_handle.rs:69]: for conversions, prefer using `try_into().expect()`, which will error if the lhs cannot convert into the rhs.

File: hubble/src/indexer/tm/block_handle.rs [hunk 2]
```diff
@@ -21,53 +25,67 @@ use crate::indexer::{
...
     }
 }
 
 impl BlockReferenceProvider for BlockHeader {
     fn block_reference(&self) -> Result<BlockReference, Report> {
         Ok(BlockReference {
-            height: self.header.height.into(),
-            hash: self.block_id.hash.to_string(),
-            timestamp: self.header.time.into(),
+            height: self.header.height.inner() as u64,
```
Comment (KaiserKarel) [inline hubble/src/indexer/tm/block_handle.rs:62]: use try_into.unwrap

File: hubble/src/indexer/tm/fetcher_client.rs [hunk 3]
```diff
@@ -221,7 +224,7 @@ impl TmFetcherClient {
 
         let pg_transactions = transactions_response
             .into_iter()
-            .filter(|tx| tx.tx_result.code.is_ok())
+            .filter(|tx| tx.tx_result.code == 0) // 0 == OK
```
Comment (KaiserKarel) [inline hubble/src/indexer/tm/fetcher_client.rs:227]: perhaps use a const: `CODE_OK = 0`, although that is a nit tbh.

Comment (qlp) [reply hubble/src/indexer/tm/fetcher_client.rs]: fixed in f40e13a0c013fb1bd8061e281ae6cd5dda72f610

**Review by cor — APPROVED** [review]

**Review by cor — APPROVED** [review]
LGTM, karels suggestions are correctly applied

File: lib/cometbft-rpc/src/lib.rs [hunk 4]
```diff
@@ -244,14 +258,14 @@ impl Client {
...
-    //     &self,
-    //     height: Option<BoundedI64<1>>,
-    // ) -> Result<TxSearchResponse, JsonRpcError> {
-    //     self.client
-    //         .request("block_results", rpc_params![height.map(|x| x.to_string())])
-    //         .await
-    // }
+    pub async fn block_results(
+        &self,
+        height: Option<NonZeroU64>,
```
Comment (benluelo) [inline lib/cometbft-rpc/src/lib.rs:263]: this should technically be `Option<BoundedI64<1>>` as it was previously

**Pull request merged**